src/CodeGeneration/ApiGenerator/Domain/Code/HighLevel/Methods/InitializerMethod.cs

```diff
@@ -28,14 +28,55 @@ public class InitializerMethod : MethodSyntaxBase
 
 		private bool IsNodesHotThreadsMethod => CsharpNames.Namespace == "Nodes" && MethodName == "HotThreads";
 
-		public string DispatchMethod => IsNodesHotThreadsMethod ? "DoNodesHotThreads" : IsCatHelpMethod ? "DoCatHelp" : IsCatMethod ? "DoCat" : "DoRequest";
+		private bool IsMultiGetMethod => CsharpNames.Namespace == "NoNamespace" && MethodName == "MultiGet";
 
-		public string DispatchGenerics => IsNodesHotThreadsMethod ? "" : IsCatMethod
-			? $"<{ArgumentType},{CsharpNames.ParametersName},{CsharpNames.RequestName.Replace("Request", "Record")}>"
-			: $"<{ArgumentType},{ResponseName}>";
+		private bool IsMultiSearchMethod => CsharpNames.Namespace == "NoNamespace" && MethodName == "MultiSearch";
 
-		public string DispatchParameters => IsCatMethod || IsNodesHotThreadsMethod
-			? "request"
-			: "request, request.RequestParameters";
+		private bool IsSourceMethod => CsharpNames.Namespace == "NoNamespace" && MethodName == "Source";
+
+		private bool IsMultiSearchTemplateMethod => CsharpNames.Namespace == "NoNamespace" && MethodName == "MultiSearchTemplate";
+
+		private bool IsSqlTranslateMethod => CsharpNames.Namespace == "Sql" && MethodName == "Translate";
+
+		public string DispatchMethod => IsNodesHotThreadsMethod
```

Comment:
This was already tackled in https://github.com/elastic/elasticsearch-net/pull/3793